lib/cartopy/mpl/geoaxes.py

```diff
@@ -1878,6 +1878,107 @@ def add_wms(self, wms, layers, wms_kwargs=None, **kwargs):
         return self.add_raster(wms, **kwargs)
 
 
+# ##### YP ####################################################################
+    def _cb_resize(self, cax, pad, size, location):
+        """
+        Returns function to automatically resize colorbar for cartopy plots
+
+        Parameters
+        ----------
+        cax : colorbar axis
+        pad: number
+            padding of colorbar from main plot
+        size: number
+            relative size (thickness) of the colorbar
+        location : str
+            location relative to the plot where to place the colorbar
+        """
+
+        def resize_colorbar(event):
+            # Tell matplotlib to re-draw everything, so that we can get the
+            # correct location from get_position.
+            mpl.pyplot.draw()
```

Comment:
We can't use pyplot in this module - this module does not want to interface with the mpl state machine (incuding pyplot.draw()). Unfortunately, that means there are quite a number hoops to jump through to arrive at the equivalent solution.